client/app/components/queries/query-editor.js

```diff
@@ -42,6 +42,10 @@ function queryEditor(QuerySnippet) {
             autoScrollEditorIntoView: true,
           },
           onLoad(editor) {
+            $scope.$on('schema-browser-select', ($event, hierarchy) => {
+              editor.session.doc.replace(editor.selection.getRange(), hierarchy.join('.'));
+            });
```

Comment:
Yep, I know. But in this case, I think, event is the best solution. The only think that may be improved - change event to be more generic, say, call it `query-editor-paste`, and send a piece of text instead of schema node. Introducing any other communication channel will be over-engineering here.

Reply:
👍 let's rename it then, makes it less coupled.

Reply:
@arikfr Done.